agent/nmap_options.py

```diff
@@ -54,11 +55,17 @@ def _set_ports_option(self):
         else:
             return []
 
-
     def _set_timing_option(self):
         """Appends the timing template option to the list of nmap options."""
         return [self.timing_template.value]
 
+    def _set_script(self):
+        """Appends the  option to the list of nmap options."""
+        if self.script is not None:
+            return ['--script', self.script]
```

Comment:
--script=<Lua scripts>: <Lua scripts> is a comma separated list of
           directories, script-files or script-categories

You need to persist the file and pass the path